evalml/tests/model_understanding_tests/test_graphs.py

```diff
@@ -1028,3 +1032,73 @@ class NotTSPipeline:
+    est_class, est_reg = tree_estimators
+    est_logit = logit_estimator
+
+    with pytest.raises(ValueError, match="Tree structure reformatting is not supported for non-Tree estimators"):
+        get_tree_data(est_logit)
+
+    formatted_ = get_tree_data(est_reg)
+    tree_ = est_reg._component_obj.tree_
+
+    assert isinstance(formatted_, OrderedDict)
+    assert formatted_['Feature'] == tree_.feature[0]
+    assert formatted_['Threshold'] == tree_.threshold[0]
+    assert all([a == b for a, b in zip(formatted_['Value'][0], tree_.value[0][0])])
+    left_child_name_ = formatted_['Left_Child']['Name']
+    right_child_name_ = formatted_['Right_Child']['Name']
+    left_child_threshold_ = formatted_['Left_Child']['Threshold']
+    right_child_threshold_ = formatted_['Right_Child']['Threshold']
+    left_child_value_ = formatted_['Left_Child']['Value']
+    right_child_value_ = formatted_['Right_Child']['Value']
+    assert left_child_name_[left_child_name_.index('_') + 1:] == str(tree_.children_left[0])
+    assert right_child_name_[right_child_name_.index('_') + 1:] == str(tree_.children_right[0])
+    assert left_child_threshold_ == tree_.threshold[tree_.children_left[0]]
+    assert right_child_threshold_ == tree_.threshold[tree_.children_right[0]]
+    # Check that the immediate left and right child of the root node have the correct values
+    assert all([a == b for a, b in zip(left_child_value_[0], tree_.value[tree_.children_left[0]][0])])
+    assert all([a == b for a, b in zip(right_child_value_[0], tree_.value[tree_.children_right[0]][0])])
+    print('--------------')
+    print(tree_.children_left)
+    print(tree_.children_right)
+    from pprint import pprint as pp
+    pp(formatted_)
+
+
+def test_visualize_decision_trees(tree_estimators, logit_estimator, tmpdir):
+    graphviz = pytest.importorskip('graphviz', reason='Skipping visualizing test because graphviz not installed')
+    est_class, est_reg = tree_estimators
+    est_logit = logit_estimator
+
+    filepath = os.path.join(str(tmpdir), 'invalid', 'path', 'test.png')
+    assert not os.path.exists(filepath)
+    with pytest.raises(ValueError, match="Specified filepath is not writeable"):
+        visualize_decision_tree(clf=est_class, filepath=filepath)
+
+    filepath = os.path.join(str(tmpdir), 'test_0.xyz')
+    with pytest.raises(ValueError, match=f"Unknown format 'xyz'. Make sure your format is one of the following: {graphviz.backend.FORMATS}"):
+        visualize_decision_tree(clf=est_class, filepath=filepath)
+
+    filepath = os.path.join(str(tmpdir), 'test_1.png')
+    with pytest.raises(ValueError, match="Tree visualizations are not supported for non-Tree estimators"):
+        visualize_decision_tree(clf=est_logit, filepath=filepath)
+
+    with pytest.raises(ValueError, match="Unknown value: '-1'. The parameter max_depth has to be a non-negative integer"):
+        visualize_decision_tree(clf=est_class, max_depth=-1, filepath=filepath)
```

Comment:
Same comment as above, could be useful to separate out this test so it's not doing too much!